src/uvw/resource.hpp

```diff
@@ -29,19 +29,19 @@ class Resource: public Emitter<T>, public std::enable_shared_from_this<T> {
           pLoop{std::move(ref)},
           resource{}
     {
-        this->template get<U>()->data = static_cast<T*>(this);
+        resource.data = static_cast<T*>(this);
     }
 
     auto parent() const noexcept {
         return pLoop->loop.get();
     }
 
-    template<typename R>
+    template<typename R = U>
```

Comment:
You don't have to pick a different name up at all. You can simply remove the default argument and add the following functions:
```
    auto get() noexcept {
        return &resource;
    }

    auto get() const noexcept {
        return static_cast<const U*>(&resource);
    }
```
Non template member methods will be chosen whenever you don't explicitly specify a template parameter, otherwise the template function will be used.
I guess this is probably the best solution to reduce both the boilerplate and the code bloating.

Could you proceed with the change?

Thank you very much.